lightningd/options.c

```diff
@@ -767,11 +768,11 @@ void handle_opts(struct lightningd *ld, int argc, char *argv[])
 
 	check_config(ld);
 
-	if (ld->portnum && tal_count(ld->wireaddrs) == 0)
+	if (ld->portnum && ld->listen && tal_count(ld->wireaddrs) == 0)
 		guess_addresses(ld);
 	else
 		log_debug(ld->log, "Not guessing addresses: %s",
-			  ld->portnum ? "manually set" : "port set to zero");
+			  (ld->portnum && ld->listen) ? "manually set" : "port set to zero");
```

Comment:
Since `ld->listen` is now involved, it may seem strange if user sets `portnum` to nonzero and then set `--offline` and then get the log message "port set to zero".  Maybe better "listening disabled"?